### PR TITLE
Rendering Extraction results on the FE w an optional Download button

### PR DESCRIPTION
**Pull request by VashishtMadhavan**
This PR:
- Displays extraction results on the FE
- Adds a download button to get results
- Breaks out extraction + retrieval code into separate components

https://www.loom.com/share/e0b3c66fcba84af3b640272b4fef82aa

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -138,35 +125,23 @@ function App() {
...
+          />
+        ) : (
+          <ExtractionView
+            items={items}
+            itemIndex={itemIndex}
+            setItemIndex={setItemIndex}
+            isLoading={isLoading}
+            handleDownloadButtonClick={handleDownloadButtonClick}
+          />
+        )}
```
Comment (andylouisqin) [inline client/src/App.js:154]: So much nicer

**Review by andylouisqin — DISMISSED** [review]
Nice! Splitting out the views will make it easier to change their UI's in the future, if we decide to have a key-value input for extraction. One thing I ran into when trying to test locally: hitting the OpenAI rate limit when making all these async requests:
```
Retrying langchain.chat_models.openai.acompletion_with_retry.<locals>._completion_with_retry in 4.0 seconds as it raised RateLimitError: Rate limit reached for default-gpt-3.5-turbo in organization org-QQxhxuNTKB0A1QcFgJmy2x1L on tokens per min. Limit: 90000 / min. Current: 88941 / min. Contact support@openai.com if you continue to have issues..
```
Run into this yourself?

File: client/src/App.js [hunk 2]
```diff
@@ -69,6 +47,14 @@ function App() {
     window.URL.revokeObjectURL(url);
   }
 
+  const handleDownloadButtonClick = (event) => {
+    fetch(`${api_endpoint}download_csv?entity_json=${query}&results_json=${JSON.stringify(items)}`)
```
Comment (andylouisqin) [inline client/src/App.js]: I think the max length of a URL is 6000 chars, which we'd hit if `items` has >60 chars per item on average (assuming unique items). Alternatively: should we turn this into a POST that has a JSON body which increases our headroom to 1GB?

**Comment by VashishtMadhavan** [comment]
> Nice! Splitting out the views will make it easier to change their UI's in the future, if we decide to have a key-value input for extraction. One thing I ran into when trying to test locally: hitting the OpenAI rate limit when making all these async requests:
> 
> ```
> Retrying langchain.chat_models.openai.acompletion_with_retry.<locals>._completion_with_retry in 4.0 seconds as it raised RateLimitError: Rate limit reached for default-gpt-3.5-turbo in organization org-QQxhxuNTKB0A1QcFgJmy2x1L on tokens per min. Limit: 90000 / min. Current: 88941 / min. Contact support@openai.com if you continue to have issues..
> ```
> 
> Run into this yourself?

Yeah I did. I think this is just a log message from langchain that signals that the request is being retried. I think all the results are still returned tho

**Pull request merged**